File: JSPM.Web/Modules/Organisation/Institute/InstituteRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace JSPM.Organisation;

[ConnectionKey("Default"), Module("Organisation"), TableName("[organisation].[Institute]")]
[DisplayName("Institute"), InstanceName("Institute")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class InstituteRow : Row<InstituteRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Institute Name"), Column("Institute Name"), Size(200), NotNull, QuickSearch, NameProperty]
    public string InstituteName
    {
        get => fields.InstituteName[this];
        set => fields.InstituteName[this] = value;
    }

    [DisplayName("Description"), Size(1000)]
    public string Description
    {
        get => fields.Description[this];
        set => fields.Description[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField InstituteName;
        public StringField Description;

    }
}

File: JSPM.Web/Modules/ServerTypes/Organisation/InstituteRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface InstituteRow {
    Id?: number;
    InstituteName?: string;
    Description?: string;
}

export abstract class InstituteRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'InstituteName';
    static readonly localTextPrefix = 'Organisation.Institute';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<InstituteRow>();
}